contracts/base/BaseLiquidityManagement.sol

```diff
@@ -0,0 +1,305 @@
+        } else if (op == LiquidityOperation.COLLECT) {
+            return abi.encode(_collectAndZeroOut(owner, range, 0, hookData, claims));
+        } else {
+            return new bytes(0);
+        }
+    }
+
+    function _modifyLiquidity(address owner, LiquidityRange memory range, int256 liquidityChange, bytes memory hookData)
+        internal
+        returns (BalanceDelta liquidityDelta, BalanceDelta totalFeesAccrued)
+    {
+        (liquidityDelta, totalFeesAccrued) = manager.modifyLiquidity(
+            range.key,
+            IPoolManager.ModifyLiquidityParams({
+                tickLower: range.tickLower,
+                tickUpper: range.tickUpper,
+                liquidityDelta: liquidityChange,
+                salt: range.key.hooks.getLiquiditySalt(owner)
+            }),
+            hookData
+        );
+    }
+
+    function _increaseLiquidity(
+        address owner,
+        LiquidityRange memory range,
+        uint256 liquidityToAdd,
+        bytes memory hookData
+    ) internal returns (BalanceDelta) {
+        // Note that the liquidityDelta includes totalFeesAccrued. The totalFeesAccrued is returned separately for accounting purposes.
+        (BalanceDelta liquidityDelta, BalanceDelta totalFeesAccrued) =
+            _modifyLiquidity(owner, range, liquidityToAdd.toInt256(), hookData);
+
+        Position storage position = positions[owner][range.toId()];
+
+        // Account for fees that were potentially collected to other users on the same range.
+        BalanceDelta callerFeesAccrued = _updateFeeGrowth(range, position);
+        BalanceDelta feesToCollect = totalFeesAccrued - callerFeesAccrued;
+        range.key.currency0.take(manager, address(this), uint128(feesToCollect.amount0()), true);
+        range.key.currency1.take(manager, address(this), uint128(feesToCollect.amount1()), true);
+
+        // the delta applied from the above actions is liquidityDelta - feesToCollect, note that the actual total delta for the caller may be different because actions can be chained
+        BalanceDelta callerDelta = liquidityDelta - feesToCollect;
+
+        // update liquidity after feeGrowth is updated
+        position.liquidity += liquidityToAdd;
+
+        // Update the tokensOwed0 and tokensOwed1 values for the caller.
+        // if callerDelta < 0, existing fees were re-invested AND net new tokens are required for the liquidity increase
+        // if callerDelta == 0, existing fees were reinvested (autocompounded)
+        // if callerDelta > 0, some but not all existing fees were used to increase liquidity. Any remainder is added to the position's owed tokens
+        if (callerDelta.amount0() > 0) {
+            position.tokensOwed0 += uint128(callerDelta.amount0());
+            range.key.currency0.take(manager, address(this), uint128(callerDelta.amount0()), true);
+            callerDelta = toBalanceDelta(0, callerDelta.amount1());
```

Comment:
possibly contentious...

for cases where some - but not all - fees are used to increase liquidity, any remainder fees are credited to the position's tokensOwed. it feels natural in the vanilla case, but in the batching world i wonder if the credit could be used towards another operation..